src/Application/Console/Commands/AnalyseCommand.php

```diff
@@ -47,14 +49,23 @@ public function __construct(Analyser $analyser, FilesRepository $filesRepository
     /**
      * Handle the given input.
      *
-     * @param  \Symfony\Component\Console\Input\InputInterface  $input
-     * @param  \Symfony\Component\Console\Output\OutputInterface  $output
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
      *
      * @return int
      */
     public function __invoke(InputInterface $input, OutputInterface $output): int
     {
-        $style = new Style($input, OutputDecorator::decorate($output));
+        $consoleOutput = $output;
+        if ($consoleOutput instanceof ConsoleOutputInterface) {
```

Comment:
On don't get this line, can you explain it to me please?

Reply:
Yeah so basically this is because of the way piping works.
When you pipe results you pipe `stdout` to something else` but it does not pipe `stderr`.
So the point is that everything we show in the console which is not the actual result, is getting parsed to `stderr` instead of `stdout`. 

        $consoleOutput = $output;
        if ($consoleOutput instanceof ConsoleOutputInterface) {
            $consoleOutput = $consoleOutput->getErrorOutput();
            $consoleOutput->setDecorated($output->isDecorated());
        }

So because we have no way of checking if `stdout` is getting piped, we always use `stderr` for data which is not relevant for the result. And then setting decorated because `stderr` as default is not decorated. 

Using `stderr` to show progress even when data is getting piped is a common approach and is also how Symfony progress bar works.

Reply:
Understood, thanks!